go/ai/gen.go

```diff
@@ -397,3 +404,24 @@ type TraceMetadata struct {
 	Paths       []*PathMetadata `json:"paths,omitempty"`
 	Timestamp   float64         `json:"timestamp,omitempty"`
 }
+
+// StartOperationFunc starts a background operation
+type StartOperationFunc[In, Out any] = func(ctx context.Context, input In) (*core.Operation[Out], error)
+
+// CheckOperationFunc checks the status of a background operation
+type CheckOperationFunc[Out any] = func(ctx context.Context, operation *core.Operation[Out]) (*core.Operation[Out], error)
+
+// CancelOperationFunc cancels a background operation
+type CancelOperationFunc[Out any] = func(ctx context.Context, operation *core.Operation[Out]) (*core.Operation[Out], error)
+
+// BackgroundModelOptions holds configuration for defining a background model
+type BackgroundModelOptions[In, Out any] struct {
```

Comment:
This actually doesn't need to be parameterized, it's always going to be `ModelRequest` for `In` and `ModelResponse` for `Out`.

Also, please move this to `generate.go`, this file is generated from the schema.